app/src/org/commcare/models/database/user/models/AndroidCaseIndexTable.java

```diff
@@ -279,4 +291,17 @@ public static String getArgumentBasedVariableSet(int number) {
         return sb.toString();
     }
 
+    public void reIndexAllCases(SqlStorage<ACase> caseStorage) {
+        db.beginTransaction();
+        try {
+            wipeTable();
+            for(ACase c : caseStorage) {
+                indexCase(c);
```

Comment:
If `db.insert()` is called on a table name that doesn't yet exist, does it create the table? Otherwise this seems like it would fail (since `wipeTable()` deletes the table entirely, rather than just wiping its contents)